search/localsearch_nqueens.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * @file
+ * @brief [Heuristic local Search](https://en.wikipedia.org/wiki/Local_search_(optimization)) algorithm.
+ * @details
+ * The LS algorithm is the basis of many methods used for optimization problems.
+ * Its system is based on the search for neighborhoods that obtain a better solution.
+ * This exercise solves the n queens problem [https://en.wikipedia.org/wiki/Eight_queens_puzzle].
+ * The rule of neighborhood applied in this file is first.improvement.
+ * @author [dhernandezl](https://github.com/dhernandezl)
+ */
+
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <ctime>
+
+#define NREINAS 8 //The number of queens on the table is the same as the dimension of the chessboard.

```

Comment:
Don't use macros, this is not good and can lead to memory leaks.
Pass them as local function parameters.

Reply:
Working¡'